src/plugins/expressions/common/executor/executor.ts

```diff
@@ -257,6 +257,21 @@ export class Executor<Context extends Record<string, unknown> = Record<string, u
     return telemetryData;
   }
 
+  migrate(ast: SerializableState, version: string) {
+    return this.walkAst(cloneDeep(ast) as ExpressionAstExpression, (fn, link) => {
+      link = fn.migrations[version](link) as ExpressionAstFunction;
```

Comment:
and we expect the migration to return ExpressionAstFunction. thats not the general case with migrations, we might be working with unknown states where the only thing we know is that its serializable. When we do have more knowledge we'll need to do the casting.

ps: open for suggestions on how to better type safe this

Reply:
Yeah the only other option that comes to mind is instead casting `fn.migrations[version] as MigrateFunction<SerializableState, ExpressionAstFunction>`, but that doesn't really make a difference here in terms of type safety, so I think we can go with the existing implementation for now